### PR TITLE
Feature - issue 11 bank transactions

### PR DESCRIPTION
**Pull request by andrei-elizaga**
Pull request for filtering: Issue #11

- Set the bank transaction view as a ModelViewSet and setup its URL for GET method `list`.
- Create filters
    - for separate `recipient` and `block__sender`
    - for both `recipient` and `block__sender` using na 'account_number` query param.
- Add django-filter to requirements.

### CODE REVIEW

File: v1/bank_transactions/filters.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+from django.db.models import Q
+
+from .models.bank_transaction import BankTransaction
+
+
+class AccountNumberFilter(filters.CharFilter):
+
+    def filter(self, qs, value):
+        queryset = None
+        if value in EMPTY_VALUES:
```
Comment (buckyroberts) [inline v1/bank_transactions/filters.py:11]: What is `EMPTY_VALUES`?

File: v1/bank_transactions/filters.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+from django_filters import rest_framework as filters
+from django.db.models import Q
+
+from .models.bank_transaction import BankTransaction
+
+
+class AccountNumberFilter(filters.CharFilter):
+
+    def filter(self, qs, value):
+        queryset = None
```
Comment (buckyroberts) [inline v1/bank_transactions/filters.py:10]: We do not need this `queryset = None`

File: v1/bank_transactions/filters.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        if value in EMPTY_VALUES:
+            return qs
+
+        kwargs = {
+        }
+        queryset = qs.filter(**kwargs)
```
Comment (buckyroberts) [inline v1/bank_transactions/filters.py:16]: is this doing anything?

File: v1/bank_transactions/tests.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
+import random
```
Comment (buckyroberts) [inline v1/bank_transactions/tests.py:1]: unused import

File: v1/bank_transactions/tests.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        Test that there are 5 instances where `bank1` is the `Block` sender.
+        """
+        params = {
+            'block__sender': self.bank1,
+        }
+        res = self.client.get(self.URL, params)
+        self.assertEqual(res.status_code, HTTPStatus.OK)
+        self.assertEqual(len(res.data), 5)
+
+    def test_filter_acount_number(self):
```
Comment (buckyroberts) [inline v1/bank_transactions/tests.py:86]: typo in `test_filter_acount_number`, should be `test_filter_account_number`

File: v1/bank_transactions/views/bank_transaction.py [hunk 6]
```diff
@@ -1,20 +1,30 @@
 from rest_framework.response import Response
```
Comment (buckyroberts) [inline v1/bank_transactions/views/bank_transaction.py:1]: Unused import

File: v1/bank_transactions/filters.py (hunk 2 above)
Comment (andrei-elizaga) [reply v1/bank_transactions/filters.py:10]: I'll removed this.  I previously used this in a different implementation but superseded with the `filter_account_number` custom filter method.

File: v1/bank_transactions/tests.py (hunk 5 above)
Comment (andrei-elizaga) [reply v1/bank_transactions/tests.py]: Fixed on f63b10c7.   Thanks for pointing it out.

File: v1/bank_transactions/views/bank_transaction.py (hunk 6 above)
Comment (andrei-elizaga) [reply v1/bank_transactions/views/bank_transaction.py]: Removed the import on c22b7058

**Pull request closed without merging**